lib/re_captcha/configuration.rb

```diff
@@ -33,7 +36,22 @@ def language_code(locale)
     end
 
     def valid?
-      !private_key.nil? && !public_key.nil?
+      reset_errors
+      set_errors
+      errors.empty?
+    end
```

Comment:
The Rails::Engine is only required if Rails is defined.  It facilitates the integration in a Rails app but it could work perfectly without it.  I don't think that's coupling.